src/client/src/tscParseLineProtocol.c

```diff
@@ -2362,24 +2350,30 @@ static int32_t parseSmlValue(TAOS_SML_KV *pKV, const char **idx,
       }
     }
 
+    if (start < cur) {
+      if (bufSize <= len + (cur - start)) {
+        bufSize *= 2;
+        value = realloc(value, bufSize);
+      }
+      memcpy(value + len, start, cur - start);  // [start, cur)
```

Comment:
Construct the result incrementally during processing its syntax.  There is no significant performance change observed here vs the old code, per a basic benchmark of performance of taosAdapter, which is the portal of ingesting data in InfluxDB line protocol.